service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -45,47 +47,48 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody Call
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest request) {
         Map<String, Object> data = request.getCaseDetails().getData();
+        CaseDetailsMap caseDetailsMap = caseDetailsMap(request.getCaseDetails());
         CaseData caseData = mapper.convertValue(data, CaseData.class);
```

Comment:
This can be changed to make use of the methods provided by `CallbackController`. First this controller would need to extend it but then it provides nice methods to get all the parts you want from the request